File: benchmarks/struct_records_inout/struct_records_inout.swift
struct Person {
    var name: String
    var age: Int
}

func makePerson(_ i: Int) -> Person {
    Person(name: "user" + String(i), age: 20 + i % 50)
}

func birthday(_ p: inout Person) {
    p.age += 1
}

func score(_ p: Person) -> Int {
    p.name.utf8.count + p.age
}

var total = 0
for i in 0..<500000 {
    var p = makePerson(i)
    birthday(&p)
    total += score(p)
}
precondition(total == 27638890, "struct_records_inout checksum")
print("assert passed, struct_records_inout is correct")

File: benchmarks/struct_records_inout/struct_records_inout.rs
struct Person {
    name: String,
    age: i64,
}

fn make_person(i: i64) -> Person {
    Person {
        name: format!("user{}", i),
        age: 20 + i % 50,
    }
}

fn birthday(p: &mut Person) {
    p.age += 1;
}

fn score(p: &Person) -> i64 {
    p.name.len() as i64 + p.age
}

fn main() {
    let mut total = 0i64;
    for i in 0..500000 {
        let mut p = make_person(i);
        birthday(&mut p);
        total += score(&p);
    }
    assert!(total == 27638890, "struct_records_inout checksum");
    println!("assert passed, struct_records_inout is correct");
}
